### ISSUE TITLE
Document spectral norm non-support on non-diagonalizable inputs.

### ISSUE THREAD

**Issue by redwrasse** [pytorch/pytorch#142840]
### 📚 The doc issue

Document spectral norm support on non-diagonalizable inputs.

The spectral norm implementation(s) rely upon a power iteration which doesn't guarantee a correct result for non-diagonalizable inputs:

- https://pytorch.org/docs/stable/generated/torch.nn.utils.spectral_norm.html
- https://pytorch.org/docs/stable/generated/torch.nn.utils.parametrizations.spectral_norm.html#torch.nn.utils.parametrizations.spectral_norm

Example:
```
import torch


m = torch.nn.Linear(2, 2)
# Example input A is non-diagonalizable, with repeated dominant eigenvalue 3.
# So expected spectral norm is sigma = 3, and returned result should be A / 3.
A = torch.tensor([[3., 3.], [0., 3.]])
with torch.no_grad():
    m.weight = torch.nn.Parameter(A)
    torch.nn.utils.parametrizations.spectral_norm(m)
    expected = A / 3
    actual = m.weight.data
    if not torch.allclose(actual, expected):
        # Spectral norm algorithm returning A / sigma computes sigma = 4.8541 instead of sigma = 3.
        print(f"expected: {expected} \ngot: {actual}")
```

### Suggest a potential alternative/fix

Spectral norm non-support on non-diagonalizable inputs should be included in the docs (perhaps as a 'warning').

**Comment by cboiteux2765** [comment]
I can try a fix if there's no assignee already.

**Comment by redwrasse** [comment]
This might also fall under docs of numerical accuracy for linear algebra operations: https://docs.pytorch.org/docs/stable/notes/numerical_accuracy.html#extremal-values-in-linalg
